geoips/commandline/geoips_command.py

```diff
@@ -303,59 +331,192 @@ def _get_registry_by_interface_and_package(self, interface, package_name):
                 interface_registry = None
         return interface_registry
 
-    def _print_plugins_short_format(self, interface, interface_registry):
-        """Print the plugins under a certain interface alongside ancillary information.
+    def _get_headers_by_command(self, args, default_headers):
+        """Retrieve the appropriate headers from a list command with the given args.
 
-        "Short Format" includes these pieces of information:
-            [GeoIPS Package, Interface, Plugin Family, Plugin Name, Relpath]
+        Headers are the top-row of the tabular output from 'list' commands. These are
+        customizable via optional args [--column, --short, --long] and we retrieve
+        the appropriate headers based on those optional arguments and the 'list' command
+        provided.
+
+        Parameters
+        ----------
+        args: argparse Argument Namespace
+            - The arguments provided to a certain list command
+
+        Returns
+        -------
+        headers: list of str
+            - The list of column headers that will be outputted
+        """
+        columns = args.columns
+        if columns:
+            headers = {}
+            for col in columns:
+                if col not in list(default_headers.keys()):
+                    err_str = f"Column header '{col}', is not a valid header. Please "
+                    err_str += "select one or more of the following keys, which "
+                    err_str += "correspond to the appropriate value:\n"
+                    err_str += f"{default_headers}"
+                    self.subcommand_parser.error(err_str)
+                headers[col] = default_headers[col]
+        else:
+            # long has been set or was defaulted to; use the default headers
+            headers = default_headers
```

Comment:
So, long is the default? Should that always be the case? Should `default_headers` be called `available_columns` and have the superset of all available columns while `--long` provides a subset of those?

This is legitimately a question, not me trying to suggest this is how I'd like it to be.

Reply:
Yes, ``--long`` is the default. I'd assume that's what we'd always want to be the case. Give the user the most information about an artifact as possible, and if it's too much, they can filter using the ``--columns`` command. We can rename that to ``available_columns`` if you'd like.

Reply:
I think I'll make which columns to display a discussion point for our visit to NRL. I do think I'd like to switch this to `available_columns` or `all_columns` or even just `columns` if it doesn't conflict with anything else.